source/application/BioXAS/BioXASAppController.cpp

```diff
@@ -459,6 +460,12 @@ QWidget* BioXASAppController::createComponentView(QObject *component)
 			componentView = new AMMotorGroupView(cryostatStage->motorGroup(), AMMotorGroupView::CompactView);
 			componentFound = true;
 		}
+
+		BioXASFilterFlipper *filterFlipper = qobject_cast<BioXASFilterFlipper*>(component);
+		if (!componentFound && filterFlipper) {
+			componentView = new BioXASFilterFlipperView(filterFlipper);
+			componentFound = true;
+		}
```

Comment:
It is really just an organizational preference. Abstract factories are a pattern of classes which do exactly what this function does (take a reference to some object, and return a constructed descendent of some base type determined from the type of the passed object). It really just depends on whether you'd personally want the code in here, or sectioned off somewhere else.